src/tools/singlejar/port.h

```diff
@@ -19,13 +19,26 @@
 // This macro is required to tell MSVC headers to also define POSIX names
 // without "_" prefix (such as "open" for "_open").
 #define _CRT_DECLARE_NONSTDC_NAMES 1
+#endif
+
 #include <fcntl.h>
 #include <io.h>
 #include <stddef.h>
 #include <stdio.h>
+#include <sys/stat.h>
 #include <sys/types.h>
 #include <time.h>
 
+#if defined(__APPLE__)
+typedef off_t off64_t;
```

Comment:
Could you add a [`static_assert`](https://en.cppreference.com/w/cpp/language/static_assert) for the `sizeof(off_t)`, just to make sure it is indeed 8 bytes?